src/community/smart-data-loader/src/main/java/org/geoserver/smartdataloader/data/SmartDataLoaderDataAccessFactory.java

```diff
@@ -0,0 +1,356 @@
+        Resource wiFolder = gdd.get(wi, "");
+        // create folder called appschema-smart inside the datastore folder
+        String pathname = wiFolder.toString() + "/" + datastoreName + "/appschema-mappings/";
+
+        // populate appschema model visitor
+        AppSchemaVisitor appSchemaDmv =
+                new AppSchemaVisitor(namespace_prefix, target_namespace, "./" + gmlFilename);
+        dm.accept(appSchemaDmv);
+        // populate gml model visitor
+        GmlSchemaVisitor gmlDmv = new GmlSchemaVisitor(namespace_prefix, target_namespace);
+        dm.accept(gmlDmv);
+
+        // save datamodel related files
+        File appschemaFile =
+                saveMappingDocument(pathname, appschemaFilename, appSchemaDmv.getDocument());
+        saveMappingDocument(pathname, gmlFilename, gmlDmv.getDocument());
+
+        // define datastore mappings and save datastore
+        Set<FeatureTypeMapping> mappings;
+        AppSchemaDataAccess dataStore;
+        URL configFileUrl = URLs.fileToUrl(appschemaFile);
+        XMLConfigDigester configReader = new XMLConfigDigester();
+        AppSchemaDataAccessDTO config = configReader.parse(configFileUrl);
+        List<String> includes = config.getIncludes();
+        for (String include : includes) {
+            params.put("url", buildIncludeUrl(configFileUrl, include));
+            createDataStore(params, true, sourceDataStoreMap, registeredAppSchemaStores);
+        }
+        mappings = AppSchemaDataAccessConfigurator.buildMappings(config, sourceDataStoreMap);
+        dataStore = new AppSchemaDataAccess(mappings, hidden);
+        registeredAppSchemaStores.add(dataStore);
+        return dataStore;
+    }
+
+    /** Helper method that allows to create the DomainModel. */
+    private DomainModel buildDomainModel(Map<String, Serializable> params, List<String> exclusions)
+            throws IOException {
+        DataStoreInfo jdbcDataStoreInfo = this.getDataStoreInfo(params);
+        String rootEntity = lookup(ROOT_ENTITY, params, String.class);
+        JDBCDataStoreFactory factory =
+                new JDBCDataStoreFactoryFinder().getFactoryFromType(jdbcDataStoreInfo.getType());
+        JDBCDataStore jdbcDataStore = null;
+        DataStoreMetadata dsm = null;
+        try {
+            // TODO need to review (since it's forcing to get a JDBC datastore based on parameters.
+            // Not sure what happen with JNDI)
+            jdbcDataStore = factory.createDataStore(jdbcDataStoreInfo.getConnectionParameters());
+            DataStoreMetadataConfig config =
+                    new JdbcDataStoreMetadataConfig(
+                            jdbcDataStore,
+                            jdbcDataStoreInfo.getConnectionParameters().get("passwd").toString());
+            dsm = (new DataStoreMetadataFactory()).getDataStoreMetadata(config);
+        } catch (SQLException e) {
+            LOGGER.log(
+                    Level.WARNING,
```

Comment:
Looks like this exception should be logged at SEVERE level, also the exception should be re-throw and operation cancelled?